extensions/common/gcp/gcp-core/src/test/java/org/eclipse/edc/gcp/common/GcpCredentialsTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *  Copyright (c) 2023 T-Systems International GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       T-Systems International GmbH
+ *
+ */
+
+package org.eclipse.edc.gcp.common;
+
+import org.eclipse.edc.connector.core.vault.InMemoryVault;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Base64;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+
+public class GcpCredentialsTest {
+
+    private GcpCredentials gcpCredential;
+
+    String accessTokenKeyName = "access_token_key_name_test";
+
+    String invalidAccessTokenKeyName = "invalid_access_token_key_name_test";
+
+    String serviceAccountKeyName = "service_account_key_name_test";
+
+    String invalidServiceAccountKeyName = "invalid_service_account_key_name_test";
+
+
+    private String serviceAccountFileInB64;
+
+    @BeforeEach
+    public void setUp() {
+        var vault = new InMemoryVault(mock(Monitor.class));
+        var tokenValue = UUID.randomUUID();
+        var serviceAccountFileInJson = "{\n" +
```

Comment:
please use multi line strings using the triple doublequotes ("""), this will avoid all the escape characters